units/generate_tasks.go

```diff
@@ -196,10 +196,9 @@ func (j *generateTasksJob) generate(ctx context.Context, t *task.Task) error {
 	start = time.Now()
 
 	// Don't use the job's context, because it's better to try finishing than to
-	// exit early after a SIGTERM from app server shutdown.
-	ctx, cancel := context.WithTimeout(context.Background(), model.DefaultParserProjectAccessTimeout)
-	defer cancel()
-	err = g.Save(ctx, j.env.Settings(), p, pp, v)
+	// exit early after a SIGTERM from app server shutdown. This should probably
+	// use a timeout rather than just the background context.
```

Comment:
Is the timeout only used for inserting tasks/variants? In which case I don't think the ticket to optimize generating tasks/variants is going to improve this situation.

Either way, this comment isn't ideal since it doesn't really explain why there isn't currently a timeout. I think if we want to keep a timeout we should just increase it, or if we think a timeout can be reintroduced after x specific work we should link the ticket for it.

Reply:
I just re-added the comment that used to be there.

I'll open a follow-up ticket to put a context deadline eventually.

Reply:
https://jira.mongodb.org/browse/EVG-18929